src/TestProblem/Hydro/CCSN/Flag_CCSN.cpp

```diff
@@ -107,36 +110,81 @@ bool Flag_Lightbulb( const int i, const int j, const int k, const int lv, const
    bool Flag = false;
 
    const double dh        = amr->dh[lv];
-   const double Center[3] = { amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2] };
    const double Pos   [3] = { amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh,
                               amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh,
                               amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh  };
-
-   const double dx = Center[0] - Pos[0];
-   const double dy = Center[1] - Pos[1];
-   const double dz = Center[2] - Pos[2];
-   const double r  = sqrt(  SQR( dx ) + SQR( dy ) + SQR( dz )  );
+   const double Center[3] = { 0.5*amr->BoxSize[0], 0.5*amr->BoxSize[1], 0.5*amr->BoxSize[2] };
+   const double dR[3]     = { Pos[0]-Center[0], Pos[1]-Center[1], Pos[2]-Center[2] };
+   const double R         = sqrt( SQR(dR[0]) + SQR(dR[1]) + SQR(dR[2]) );
 
    const real (*Rho )[PS1][PS1] = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DENS];
 
 
 // TODO: fine-tune the criteria
 // (1) always refined to highest level in the region with r < 30 km
-   if ( r * UNIT_L < 3e6 )
+   if      ( R * UNIT_L < 3e6 )
       Flag = true;
 
-   else
-   {
-//    (2-a) density is larger than the threshold in Input__Flag_User
-      if ( Rho[k][j][i] < Threshold[0] )   return false;
+// (2) density is larger than the threshold in Input__Flag_User
+   else if ( Rho[k][j][i] < Threshold[0] )
+      Flag = false;
```

Comment:
The usage `CCSN_MaxRefine_RadFac` is similar to `CCSN_Max_AngRes` but not totally the same. If you want to remove `CCSN_MaxRefine_RadFac`, then this criterion on density is no longer needed.

Reply:
I simply removed `CCSN_MaxRefine_RadFac` and corresponding density criterion.